### DISCUSSION TITLE
Oil level sensor with 2024 API

### DISCUSSION THREAD

**Discussion by TheDelta** [audiconnect/audi_connect_ha#276] [General]
I'm not even sure if the oil level sensor ever returned something reliable but it was available (and also used on the example card).
Does someone actually experienced a change on the oil level?

Couldn't find the data in the new API yet, so maybe it was removed entirely or moved somewhere else.
I can't also find anything within the Audi Connect App. Only on the website where I tells me what oil type I should use.

(Requires Token Authentication:) GET API request: `https://cwat-oil-norm-service-prod.apps.mega.cariad.cloud/api/oil-norm/$VIN?language=de-DE`

JSON Response with HTTP 200:
```json
{
    "vin": "$VIN",
    "modelYear": "N/A",
    "unit": "Liter",
    "fix": {
        "amount": 7.1,
        "oilNorm": [
            "VW 504 00 (0W-30)"
        ]
    },
    "flexible": {
        "amount": 7.1,
        "oilNorm": [
            "VW 504 00 (0W-30)"
        ]
    },
    "dealerId": null
}
```

**Comment by Kolbi** [comment]
I need to check but I guess the new api:
https://emea.bff.cariad.digital/vehicle/v1/vehicles/{vin}/selectivestatus?jobs=activeVentilation,auxiliaryHeating,batteryChargingCare,batterySupport,charging,chargingTimers,chargingProfiles,climatisation,climatisationTimers,departureProfiles,fuelStatus,honkAndFlash,hybridCarAuxiliaryHeating,userCapabilities,departureTimers,lvBattery,readiness,measurements,**oilLevel**,vehicleHealthInspection,access,vehicleLights,vehicleHealthWarnings

has an answer at least:
`"oilLevel":{
      "oilLevelStatus":{
         "value":{
            "carCapturedTimestamp":,
            **"value":true**
         }
      }
   },`
           
And true currently should give back to HASSIO 100%. :)  

See: https://github.com/audiconnect/audi_connect_ha/blob/81b78ca59747707eca7a496e28a89dea8238c381/custom_components/audiconnect/audi_connect_account.py#L835

But didn't checked it yet it code is working currently.

**Comment by sramka** [comment]
mobileAPI: https://emea.bff.cariad.digital/vehicle/v1/vehicles/{vin}/selectivestatus
only returns value "true" for me:

```
    "oilLevelStatus": {
      "value": {
        "carCapturedTimestamp": "2024-03-12T14:46:44Z",
        "value": true
      }
    }

```
while the webapp API: https://userinfo.my.audi.com/bvh/v1
returns also the actual % that I can also see in car (it fluctuates):

```
            "oilLevel": {
                "value": 100,
                "unit": "%"
            },
            "oilLevelSupported": true,

```
